src/modules/socket.ts

```diff
@@ -2,147 +2,225 @@ import { io, Socket } from "socket.io-client";
-    this.socket.on("connect", () => {
-      console.log("Socket is connected");
-    });
-    this.socket.on("disconnect", () => {
-      console.log("Socket is disconnected");
-    });
-    this.socket.on(
-      SocketEventType.RECEIVE_VIDEO_EVENT,
-      (message: EventInfo) => {
-        console.log("Received video event from the server: ", message);
-        switch (message.event) {
-          case "play":
-            this.video.play();
-            console.log("Video is played!");
-            break;
-          case "pause":
-            this.video.pause();
-            console.log("Video is paused!");
-            break;
-          case "seeked":
-            //this.video.currentTime = message.currentTime;
-            console.log("Video is seeked!", message.currentTime);
-            break;
-        }
-      }
-    );
-    this.socket.on(SocketEventType.PERMISSIONS, (message: any) => {
-      console.log("Received video event from the server: ", message);
-      this.permissions = message.permissions;
-    });
-  };
+    openConnection = () => {
+        this.socket = io(this.serverUrl);
+        this.socket.on("connect", () => {
+            console.log("Socket is connected");
+        });
+        this.socket.on("disconnect", () => {
+            console.log("Socket is disconnected");
+        });
+        this.socket.on(
+            SocketEventType.RECEIVE_VIDEO_EVENT,
+            (message: EventInfo) => {
+                this.seek = true;
+                console.log("Received video event from the server: ", message);
+                switch (message.event) {
+                    case "play":
+                        if (Math.abs(this.video.currentTime - message.currentTime) > 0.5 ) {
+                            const time1 = message.currentTime * 1000;
+                            const seekCode1 = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                                const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                                player.seek(${time1});`;
+                            document.documentElement.setAttribute(
+                                "onreset",
+                                seekCode1
+                            );
+                            document.documentElement.dispatchEvent(
+                                new CustomEvent("reset")
+                            );
+                        }
 
-  sendMessage = (message: string) => {
-    this.checkForErrors();
-    this.socket.emit(SocketEventType.MESSAGE, message, (response: string) => {
-      console.log("Response: " + response);
-    });
-  };
+                        console.log(this.video);
+                        const playCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                        const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
```

Comment:
This is the most important part of the change, when we receive the event, we use a netflix API call to control the video, if normal HTML function would be used, then the player would break.